flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -901,7 +901,12 @@ public void releaseOrPromotePartitions(
         try {
             partitionTracker.stopTrackingAndReleaseJobPartitions(partitionToRelease);
             partitionTracker.promoteJobPartitions(partitionsToPromote);
-
+            if (establishedResourceManagerConnection != null) {
+                establishedResourceManagerConnection
+                        .getResourceManagerGateway()
+                        .reportClusterPartitions(
```

Comment:
Why we need to report cluster partitions here? Does it means to add the cluster partition to the ClusterPartitionManager? What would happen if we do this add the JM side directly?

Reply:
The cluster partition status is currently encapsulated in the heartbeat payload from task executors to RM. The cluster partition may not be available at the RM when the job is finished so we should report the cluster partition status as soon as the partition is promoted to cluster partition.

To do this in task executor, we reuse the logic to report the status of cluster partition, which can make it much simpler than reimplementing this in JM side. Thus, I prefer doing this in task executor. What do you think?